src/Tests/dotnet.Tests/EnvironmentVariableNamesTests.cs

```diff
@@ -0,0 +1,78 @@
+    [InlineData("os.1.2-loongarch64", Architecture.LoongArch64)]
+    [InlineData("os.1.2-armv6", Architecture.Armv6)]
+    [InlineData("os.1.2-ppc64le", Architecture.Ppc64le)]
+    [InlineData("os.1.2-lOOngaRch64", Architecture.LoongArch64)] // case-insensitive
+    [InlineData("os-x86", Architecture.X86)]
+    [InlineData("-x86", Architecture.X86)]
+    [InlineData("-x86-", Architecture.X86)]
+    public static void TryParseArchitecture(string rid, Architecture expected)
+    {
+        Assert.True(EnvironmentVariableNames.TryParseArchitecture(rid, out var actual));
+        Assert.Equal(expected, actual);
+
+        Assert.True(EnvironmentVariableNames.TryParseArchitecture(rid + "-xyz", out actual));
+        Assert.Equal(expected, actual);
+    }
+
+    [Theory]
+    [InlineData("")]
+    [InlineData("-")]
+    [InlineData("--")]
+    [InlineData("---")]
+    [InlineData("x86")]
+    [InlineData("os")]
+    [InlineData("os.")]
+    [InlineData("os.1")]
+    [InlineData("os.1.2")]
+    [InlineData("os.1.2-")]
+    [InlineData("os.1.2--")]
+    [InlineData("os.1.2-unknown")]
+    [InlineData("os.1.2-unknown-")]
+    [InlineData("os.1.2-unknown-x")]
+    [InlineData("os.1.2-armel")] // currently not defined
+    public static void TryParseArchitecture_Invalid(string rid)
+    {
+        Assert.False(EnvironmentVariableNames.TryParseArchitecture(rid, out _));
+    }
+
+    [Theory]
+    [InlineData("os-unknown", null, Architecture.X86, true, "DOTNET_ROOT")]
+    [InlineData("os-unknown", null, Architecture.X86, false, "DOTNET_ROOT(x86)")]
+    [InlineData("os-unknown", "v5.0", Architecture.X86, true, "DOTNET_ROOT")]
+    [InlineData("os-unknown", "v5.0", Architecture.X86, false, "DOTNET_ROOT(x86)")]
+    [InlineData("os-unknown", "v6.0", Architecture.Wasm, true, "DOTNET_ROOT_WASM")]
+    [InlineData("os-unknown", "v6.0", Architecture.Wasm, false, "DOTNET_ROOT_WASM")]
+    [InlineData("os-x86", null, Architecture.X86, true, "DOTNET_ROOT")]
+    [InlineData("os-x86", null, Architecture.X86, false, "DOTNET_ROOT(x86)")]
+    [InlineData("os-x86", "v5.0", Architecture.X86, true, "DOTNET_ROOT")]
+    [InlineData("os-x86", "v5.0", Architecture.X86, false, "DOTNET_ROOT(x86)")]
+    [InlineData("os-x86", "v6.0", Architecture.X86, true, "DOTNET_ROOT_X86")]
+    [InlineData("os-x86", "v6.0", Architecture.X86, false, "DOTNET_ROOT_X86")]
+    [InlineData("os-x64", "v6.0", Architecture.X86, false, null)]
+    public static void TryGetDotNetRootVariableName(string rid, string frameworkVersion, Architecture currentArchitecture, bool is64bit, string expected)
+    {
+        var parsedVersion = EnvironmentVariableNames.TryParseTargetFrameworkVersion(frameworkVersion);
+        Assert.Equal(expected, EnvironmentVariableNames.TryGetDotNetRootVariableNameImpl(rid, "", parsedVersion, currentArchitecture, is64bit));
+        Assert.Equal(expected, EnvironmentVariableNames.TryGetDotNetRootVariableNameImpl("", rid, parsedVersion, currentArchitecture, is64bit));
+    }
```

Comment:
Add a case where target rid is passed in, and default rid is something different like `os-armv6`.